services/src/api/model/responses.rs

```diff
@@ -0,0 +1,169 @@
+use crate::handlers::ErrorResponse;
+use utoipa::ToResponse;
+
+#[derive(ToResponse)]
+#[response(description = "Media type of application/json is expected", example = json!({
+    "error": "UnsupportedMediaType",
+    "message": "Unsupported content type header."
+}))]
+pub struct UnsupportedMediaTypeForJsonResponse(ErrorResponse);
+
+#[derive(ToResponse)]
+#[response(description = "Payload too large", examples(
+    ("Unknown payload size" = (value = json!({
+        "error": "Overflow",
+        "message": "JSON payload has exceeded limit (2097152 bytes)."
+    }))),
+    ("Known payload size" = (value = json!({
+        "error": "Overflow",
+        "message": "JSON payload (XXX bytes) is larger than allowed (limit: 2097152 bytes)."
+    })))
+))]
+pub struct PayloadTooLargeResponse(ErrorResponse);
+
+#[derive(ToResponse)]
+#[response(description = "Id of generated resource", example = json!({
+    "id": "36574dc3-560a-4b09-9d22-d5945f2b8093"
+}))]
+pub struct IdResponse {
+    pub id: String,
+}
+
+#[derive(ToResponse)]
+#[response(description = "Authorization failed", examples(
+    ("Authorization Header is missing" = (value = json!({
+        "error": "MissingAuthorizationHeader",
+        "message": "Header with authorization token not provided."
+    }))),
+    ("Authorization Scheme other than Bearer is used" = (value = json!({
+        "error": "InvalidAuthorizationScheme",
+        "message": "Authentication scheme must be Bearer."
+    }))),
+    ("Provided auth token has an invalid format" = (value = json!({
+        "error": "InvalidUuid",
+        "message": "Identifier does not have the right format."
+    }))),
+    ("Auth token does not correspond to an admin" = (value = json!({
+        "error": "InvalidAdminToken",
+        "message": "Invalid admin token"
+    }))),
+    ("Session id is invalid" = (value = json!({
+        "error": "InvalidSession",
+        "message": "The session id is invalid."
+    }))),
+))]
+pub struct UnauthorizedAdminResponse(ErrorResponse);
+
+#[derive(ToResponse)]
+#[response(description = "Authorization failed", examples(
+    ("Authorization Header is missing" = (value = json!({
+        "error": "MissingAuthorizationHeader",
+        "message": "Header with authorization token not provided."
+    }))),
+    ("Authorization Scheme other than Bearer is used" = (value = json!({
+        "error": "InvalidAuthorizationScheme",
+        "message": "Authentication scheme must be Bearer."
+    }))),
+    ("Provided auth token has an invalid format" = (value = json!({
+        "error": "InvalidUuid",
+        "message": "Identifier does not have the right format."
+    }))),
+    ("Session id is invalid" = (value = json!({
+        "error": "InvalidSession",
+        "message": "The session id is invalid."
+    })))
+))]
+pub struct UnauthorizedUserResponse(ErrorResponse);
+
+/*("Operation can only be executed by an admin" = (value = json!({
```

Comment:
why is this commented out?

Reply:
I found out that this is a possible authorization error. I copied it to the other authorization errors so I will not miss it out when I will get to handler which can return that. I will delete it later on.

Reply:
[d79292d](https://github.com/geo-engine/geoengine/pull/751/commits/d79292d69a885193432a939160cc92c184d29ef1)